R/assign.R

```diff
@@ -2,18 +2,51 @@
 sdtm_assign <- function(raw_dat,
                         raw_var,
                         tgt_var,
-                        tgt_dat = raw_dat,
-                        by = NULL,
+                        tgt_dat,
+                        by_var = NULL,
                         ct = NULL,
                         cl = NULL) {
 
   # TODO: Assertions.
 
-  raw_dat |>
-    dplyr::right_join(y = tgt_dat, by = by) |>
-    dplyr::mutate("{tgt_var}" := ct_map(!!rlang::sym(raw_var), ct = ct, cl = cl)) |>
-    dplyr::select(-rlang::sym(raw_var)) |>
-    dplyr::relocate(tgt_var, .after = dplyr::last_col())
+  # When target dataset and  by_var variables are provided,
+  # the tar_var is added to the input target dataset.
+  # This if block deals when ct is not provided.
+  if((!is.null(by_var) & is.null(ct))){
```

Comment:
When the predicates are scalars it is good practice to use `&&` instead of `&` (vectorized form), as it conveys its intent more clearly.